packages/java/endpoint/src/main/java/dev/hilla/EndpointControllerConfiguration.java

```diff
@@ -55,6 +55,119 @@ public EndpointControllerConfiguration(
+     *
+     * @param servletContext
+     *            the servlet context
+     *
+     * @return the default bean
+     */
+    @Bean
+    public CsrfChecker csrfChecker(ServletContext servletContext) {
+        return new CsrfChecker(servletContext);
+    }
+
+    /**
+     * Registers the endpoint invoker.
+     *
+     * @param applicationContext
+     *            Spring context to extract beans annotated with
+     *            {@link Endpoint} from
+     * @param endpointMapperFactory
+     *            optional bean to override the default
+     *            {@link JacksonObjectMapperFactory} that is used for
+     *            serializing and deserializing request and response bodies Use
+     *            {@link EndpointController#ENDPOINT_MAPPER_FACTORY_BEAN_QUALIFIER}
+     *            qualifier to override the mapper.
+     * @param explicitNullableTypeChecker
+     *            the method parameter and return value type checker to verify
+     *            that null values are explicit
+     * @param servletContext
+     *            the servlet context
+     * @param endpointRegistry
+     *            the registry used to store endpoint information
+     *
+     * @return the endpoint invoker
+     */
+    @Bean
+    public EndpointInvoker endpointInvoker(
+            ApplicationContext applicationContext,
+            @Autowired(required = false) @Qualifier(EndpointController.ENDPOINT_MAPPER_FACTORY_BEAN_QUALIFIER) JacksonObjectMapperFactory endpointMapperFactory,
+            ExplicitNullableTypeChecker explicitNullableTypeChecker,
+            ServletContext servletContext, EndpointRegistry endpointRegistry) {
+        return new EndpointInvoker(applicationContext, endpointMapperFactory,
+                explicitNullableTypeChecker, servletContext, endpointRegistry);
+    }
+
+    /**
+     * Registers an endpoint name checker responsible for validating the
+     * endpoint names.
+     *
+     * @return the endpoint name checker
+     */
+    @Bean
+    public EndpointNameChecker endpointNameChecker() {
+        return new EndpointNameChecker();
+    }
+
+    /**
+     * Registers the endpoint registry.
+     *
+     * @param endpointNameChecker
+     *            the name checker to use
+     * @return the endpoint registry
+     */
+    @Bean
+    public EndpointRegistry endpointRegistry(
+            EndpointNameChecker endpointNameChecker) {
+        return new EndpointRegistry(endpointNameChecker);
+    }
+
+    /**
+     * Registers endpoint utility methods.
+     *
+     * @return the endpoint util class
+     */
+    @Bean
+    public EndpointUtil endpointUtil() {
+        return new EndpointUtil();
+    }
+
+    /**
+     * Registers a {@link ExplicitNullableTypeChecker} bean instance.
+     *
+     * @return the explicit nullable type checker
+     */
+    @Bean
+    public ExplicitNullableTypeChecker typeChecker() {
+        return new ExplicitNullableTypeChecker();
+    }
+
```

Comment:
Those big code blocks moving in the same class make it hard to spot changes

Reply:
Oops, sorry, I tried to maintain the original order. Should I probably put it back?